kubernetes/utils/create_from_yaml.py

```diff
@@ -23,51 +23,85 @@
     :param str pretty: If 'true', then the output is pretty printed.
     :param str dry_run: When present, indicates that modifications should not be persisted. An invalid or unrecognized dryRun directive will result in an error response and no further processing of the request. Valid values are: - All: all dry run stages will be processed
     """
 
+    k8s_api_all = []
     with open(path.abspath(yaml_file)) as f:
-        yml_object = yaml.load(f)
-        # TODO: case of yaml file containing multiple objects
-        group, _, version = yml_object["apiVersion"].partition("/")
-        if version == "":
-            version = group
-            group = "core"
-        # Take care for the case e.g. api_type is "apiextensions.k8s.io"
-        # Only replace the last instance
-        group = "".join(group.rsplit(".k8s.io", 1))
-        fcn_to_call = "{0}{1}Api".format(group.capitalize(),
-                                         version.capitalize())
-        k8s_api = getattr(client, fcn_to_call)(k8s_client)
-        # Replace CamelCased action_type into snake_case
-        kind = yml_object["kind"]
-        kind = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', kind)
-        kind = re.sub('([a-z0-9])([A-Z])', r'\1_\2', kind).lower()
-        # Decide which namespace we are going to put the object in,
-        # if any
-        if "namespace" in yml_object["metadata"]:
-            namespace = yml_object["metadata"]["namespace"]
-        else:
-            namespace = "default"
-        # Expect the user to create namespaced objects more often
-        if hasattr(k8s_api, "create_namespaced_{0}".format(kind)):
-            resp = getattr(k8s_api, "create_namespaced_{0}".format(kind))(
-                body=yml_object, namespace=namespace, **kwargs)
-        else:
-            resp = getattr(k8s_api, "create_{0}".format(kind))(
-                body=yml_object, **kwargs)
-        if verbose:
-            print("{0} created. status='{1}'".format(kind, str(resp.status)))
-        return k8s_api
+        yml_document_all = yaml.load_all(f)
+        # Load all documents from a single YAML file
+        for yml_document in yml_document_all:
+            # If it is a list type, will need to iterate its items
+            if "List" in yml_document["kind"]:
+                # Could be "List" or "Pod/Service/...List"
+                # This is a list type. iterate within its items
+                for yml_object in yml_document["items"]:
+                    k8s_api_all.append(create_from_yaml_single_item(
+                        k8s_client, yml_object, verbose, **kwargs))
+            else:
+                # This is a single object. Call the single item method
+                k8s_api_all.append(create_from_yaml_single_item(
+                    k8s_client, yml_document, verbose, **kwargs))
+    if output_list is False:
+        if len(k8s_api_all) == 1:
+            return k8s_api_all[0]
```

Comment:
Should this be returning the first item when `output_list is False` or `len(k8s_api_all) == 1`?

Reply:
See above - I’m think about the case when the user do not want to check the return type and just want to iterate a list.